yb-voyager/src/migassessment/sizing.go

```diff
@@ -115,416 +172,503 @@ func SizingAssessment() error {
 		calculateTimeTakenAndParallelJobsForImport(SHARDED_LOAD_TIME_TABLE, shardedObjects,
-			coresToUse.numCores.Float64, coresToUse.memPerCore.Float64)
+			finalSizingRecommendation.VCPUsPerInstance, finalSizingRecommendation.MemoryPerCore)
 	if err != nil {
 		SizingReport.FailureReasoning = fmt.Sprintf("calculate time taken for sharded data import: %v", err)
 		return fmt.Errorf("calculate time taken for sharded data import: %w", err)
 	}
+
 	sizingRecommendation := &SizingRecommendation{
 		ColocatedTables:                 fetchObjectNames(colocatedObjects),
 		ShardedTables:                   fetchObjectNames(shardedObjects),
-		VCPUsPerInstance:                coresToUse.numCores.Float64,
-		MemoryPerInstance:               coresToUse.numCores.Float64 * coresToUse.memPerCore.Float64,
-		NumNodes:                        numNodes,
-		OptimalSelectConnectionsPerNode: optimalSelectConnections,
-		OptimalInsertConnectionsPerNode: optimalInsertConnections,
+		VCPUsPerInstance:                finalSizingRecommendation.VCPUsPerInstance,
+		MemoryPerInstance:               finalSizingRecommendation.VCPUsPerInstance * finalSizingRecommendation.MemoryPerCore,
+		NumNodes:                        finalSizingRecommendation.NumNodes,
+		OptimalSelectConnectionsPerNode: finalSizingRecommendation.OptimalSelectConnectionsPerNode,
+		OptimalInsertConnectionsPerNode: finalSizingRecommendation.OptimalInsertConnectionsPerNode,
 		ParallelVoyagerJobs:             math.Min(float64(parallelVoyagerJobsColocated), float64(parallelVoyagerJobsSharded)),
-		ColocatedReasoning:              reasoning,
+		ColocatedReasoning:              getReasoning(finalSizingRecommendation, shardedObjects, colocatedObjects),
 		EstimatedTimeInMinForImport:     importTimeForColocatedObjects + importTimeForShardedObjects,
 	}
 	SizingReport.SizingRecommendation = *sizingRecommendation
 
 	return nil
 }
 
-func fetchObjectNames(dbObjects []SourceDBMetadata) []string {
-	var objectNames []string
-	for _, dbObject := range dbObjects {
-		objectNames = append(objectNames, dbObject.SchemaName+"."+dbObject.ObjectName)
+/*
+pickBestRecommendation selects the best recommendation from a map of recommendations based on certain criteria.
+Parameters:
+  - recommendation: A map where the key is the number of vCPUs per instance and the value is an IntermediateRecommendation struct.
+
+Returns:
+  - The best IntermediateRecommendation based on the defined criteria.
+*/
+func pickBestRecommendation(recommendation map[int]IntermediateRecommendation) IntermediateRecommendation {
+	// find the one with least number of nodes
+	var minCores int = math.MaxUint32
+	var finalRecommendation IntermediateRecommendation
+	var foundRecommendation bool = false
+	var maxCores int = math.MinInt32
+
+	// Iterate over each recommendation
+	for _, rec := range recommendation {
+		// Update maxCores with the maximum number of vCPUs per instance across recommendations
```

Comment:
mention that this is to handle the scenario where none of the setups is able to satisfy the criteria

Reply:
done.